server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,273 @@
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "analytics":{
+ *              "jvm": 0.4,
+ *              "mpde": "enforced",
+ *              "_id": "fafjafjkaf9ag8a9ga9g7ag0aagaga"
+ *      }
+ * }
+ */
+@ExperimentalApi
+public class ResourceLimitGroup extends AbstractDiffable<ResourceLimitGroup> implements ToXContentObject {
+
+    public static final int MAX_CHARS_ALLOWED_IN_NAME = 50;
+    private final String name;
+    private final String _id;
+    private final ResourceLimitGroupMode mode;
+    private final Map<String, Object> resourceLimits;
+
+    // list of resources that are allowed to be present in the ResourceLimitGroupSchema
+    public static final List<String> ALLOWED_RESOURCES = List.of("jvm");
+
+    public ResourceLimitGroup(String name, String _id, ResourceLimitGroupMode mode, Map<String, Object> resourceLimits) {
+        Objects.requireNonNull(name, "ResourceLimitGroup.name can't be null");
+        Objects.requireNonNull(resourceLimits, "ResourceLimitGroup.resourceLimits can't be null");
+        Objects.requireNonNull(mode, "ResourceLimitGroup.mode can't be null");
+        Objects.requireNonNull(_id, "ResourceLimitGroup._id can't be null");
+
+        if (name.length() > MAX_CHARS_ALLOWED_IN_NAME) {
+            throw new IllegalArgumentException("ResourceLimitGroup.name shouldn't be more than 50 chars long");
+        }
+
+        if (resourceLimits.isEmpty()) {
+            throw new IllegalArgumentException("ResourceLimitGroup.resourceLimits should at least have 1 resource limit");
+        }
+        validateResourceLimits(resourceLimits);
+
+        this.name = name;
+        this._id = _id;
+        this.mode = mode;
+        this.resourceLimits = resourceLimits;
+    }
+
+    public ResourceLimitGroup(StreamInput in) throws IOException {
+        this(in.readString(), in.readString(), ResourceLimitGroupMode.fromName(in.readString()), in.readMap());
+    }
+
+    /**
+     * Write this into the {@linkplain StreamOutput}.
+     *
+     * @param out
+     */
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(name);
+        out.writeString(_id);
+        out.writeString(mode.getName());
+        out.writeMap(resourceLimits);
+    }
+
+    private void validateResourceLimits(Map<String, Object> resourceLimits) {
+        for (Map.Entry<String, Object> resource : resourceLimits.entrySet()) {
+            String resourceName = resource.getKey();
+            Double threshold = (Double) resource.getValue();
+            Objects.requireNonNull(resourceName, "resourceName can't be null");
+            Objects.requireNonNull(threshold, "resource value can't be null");
+
+            if (Double.compare(threshold, 1.0) > 0) {
```

Comment:
nit: do we need a constant for this?

Reply:
This is standard comparator return values in cases where first arg in the comparator is greater than second one